packages/firestore/e2e/Bundle/namedQuery.e2e.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2016-present Invertase Limited & Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this library except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+const { wipe, httpGet } = require('../helpers');
+const BUNDLE_HOST = 'api.rnfirebase.io';
+const BUNDLE_PATH = '/firestore/bundle';
+const QUERY_NAME = 'named-bundle-test';
+const COLLECTION = 'firestore-bundle-tests';
+
+describe('firestore().namedQuery()', function () {
+  beforeEach(async function () {
+    return await wipe();
+  });
+  async function prepareBundle() {
+    const bundleString = await httpGet(BUNDLE_HOST, BUNDLE_PATH);
```

Comment:
This not a full review - apologies that I have not done so already, just finishing up with some google admob stuff and shifting focus back over here to firebase. I can say this is an immense effort - I forgot that bundles implied named queries at the same time, and the test fixtures are non-trivial too!

One thing the e2e testing makes me think about is that most of our tests work offline, in fact I try to do all testing using the firebase emulator suite, and I *think* this one could work that way as well if the bundle string was a fixed resource inside helpers.js ? Is that possible? If not, the auth/e2e/helpers.js already has a REST API httpGet implementation and at least those could perhaps be abstracted to app/e2e/helpers and just used by reference instead of implementing the get twice ?

Reply:
Thank you for a response! I don't want push you to hurry. I just think I need to say this before I finish the PR 😀


> think this one could work that way as well if the bundle string was a fixed resource inside helpers.js

I agree with your though, since data bundles are actually just static text file. I added a bundle resource inside the  helpers.